deepchem/models/torch_models/gan.py

```diff
@@ -454,3 +454,326 @@ def _list_or_tensor(inputs):
+
+    def fit_gan(self,
+                batches,
+                generator_steps=1,
+                max_checkpoints_to_keep=5,
+                checkpoint_interval=1000,
+                restore=False):
+        """Train this model on data.
+
+        Parameters
+        ----------
+        batches: iterable
+            batches of data to train the discriminator on, each represented as a dict
+            that maps Inputs to values.  It should specify values for all members of
+            data_inputs and conditional_inputs.
+        generator_steps: float
+            the number of training steps to perform for the generator for each batch.
+            This can be used to adjust the ratio of training steps for the generator
+            and discriminator.  For example, 2.0 will perform two training steps for
+            every batch, while 0.5 will only perform one training step for every two
+            batches.
+        max_checkpoints_to_keep: int
+            the maximum number of checkpoints to keep.  Older checkpoints are discarded.
+        checkpoint_interval: int
+            the frequency at which to write checkpoints, measured in batches.  Set
+            this to 0 to disable automatic checkpointing.
+        restore: bool
+            if True, restore the model from the most recent checkpoint before training
+            it.
+        """
+        self._ensure_built()
+
+        gen_train_fraction = 0.0
+        discrim_error = 0.0
+        gen_error = 0.0
+        discrim_average_steps = 0
+        gen_average_steps = 0
+        time1 = time.time()
+
+        for feed_dict in batches:
+            # Every call to fit_generator() will increment global_step, but we only
+            # want it to get incremented once for the entire batch, so record the
+            # value and keep resetting it.
+
+            global_step = self.get_global_step()
+
+            # Train the discriminator.
+
+            inputs = [self.get_noise_batch(self.batch_size)]
+
+            temp_data = []
+            for shape in self.get_data_input_shapes():
+                temp_data.append(
+                    torch.randn(size=[self.batch_size] + list(shape[1:])))
+
+            temp_cond = []
+            for shape in self.get_conditional_input_shapes():
+                temp_cond.append(
+                    torch.randn(size=[self.batch_size] + list(shape[1:])))
+            # temp_data = torch.randn(self.get_data_input_shapes()[0])
+            # inputs=(temp_noise,temp_data,temp_cond)
+            # print("\n\n\ninputs OG:", inputs)
+            inputs += [*temp_data, *temp_cond]
+
+            # # Torch Tensor Hashing Issue here
+            # for input in self.data_input_layers:
+            #     print("Input: ", input.data)
+            #     inputs.append(feed_dict[input])
+            # for input in self.conditional_input_layers:
+            #     inputs.append(feed_dict[input])
```

Comment:
The torch tensor cannot be referenced here although it has the same value due to the way torch deals with hashing

Reply:
I don't understand what you mean here. Please explain in more detail. What is the conceptual issue? Why does every GAN implementation in pytorch not have this problem?

Reply:
This issue is occurring due to the way we are feeding in the data, if we fed the data in a different way, it could be avoided, but it would break the similarity between the TF and the Torch API